src/shared/directives/substitutor/services/placeholder.service.ts

```diff
@@ -116,57 +123,58 @@ ___
             this.fieldIdToSubstitute = '';
         }
 
-        private appendOriginalStringIfCollectionItemAsPlaceholder() {
+        private appendOriginalStringIfCollectionItemAsPlaceholder(): void {
             if (this.collectionItemIndex < this.numberCollectionItemsAsPlaceholder - 1) {
                 this.stringToResolve += PlaceholderSubstitutor.NEW_LINE + this.originalStringToResolve;
                 this.collectionItemIndex += 1;
                 this.resolvedFormValues[this.collectionItemIndex] = {};
             }
         }
 
-        private setStartCollecting() {
+        private setStartCollecting(): void {
             this.isCollecting = true;
             this.startSubstitutionIndex = this.scanIndex;
         }
 
-        private appendCharacter() {
+        private appendCharacter(): void {
             this.fieldIdToSubstitute += this.stringToResolve.charAt(this.scanIndex);
         }
 
-        private isMatchingPlaceholderPattern() {
-            return this.fieldIdToSubstitute.match(PlaceholderSubstitutor.PLACEHOLDER_CONTENT_PATTERN);
+        private isMatchingPlaceholderPattern(): boolean {
+            return !!this.fieldIdToSubstitute.match(PlaceholderSubstitutor.PLACEHOLDER_CONTENT_PATTERN);
         }
 
-        private isFieldIdInFormFields() {
+        private isFieldIdInFormFields(): boolean {
             return this.getFieldValue() !== undefined;
         }
 
-        private isFieldIdToSubstituteReferringItself() {
-            let value = this.getSubstitutionValueOrEmpty();
-            return '${'.concat(this.fieldIdToSubstitute).concat('}') === value;
+        private isFieldIdToSubstituteReferringItself(): boolean {
+            const placeholder = PlaceholderSubstitutor.wrapPlaceholder(this.fieldIdToSubstitute);
+            const value = this.getSubstitutionValueOrEmpty();
+            return placeholder === value;
         }
 
-        private getSubstitutionValueLengthOrZero() {
+        private getSubstitutionValueLengthOrZero(): number {
             return this.pageFormFields[this.fieldIdToSubstitute] ? this.getSubstitutionValueOrEmpty().toString().length : 0;
         }
 
-        private getFieldValue() {
+        private getFieldValue(): any {
```

Comment:
Just checking that there's no more suitable match than `any` for the return type.

Reply:
This kind of makes sense as `any` in this instance as it could (off the top of my head) be any of:
`string | number | object | string[] | object[]`

I personally think that looks less pleasant. Would adding a descriptive comment on the method help?

Reply:
Agreed - perhaps a method comment would be useful to explain the above

Reply:
Done.